src/Harmony/StakingSigner.cpp

```diff
@@ -0,0 +1,320 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "StakingSigner.h"
+#include "../Ethereum/RLP.h"
+#include "../HexCoding.h"
+
+using namespace TW;
+using namespace TW::Harmony;
+
+template <typename Directive>
+Proto::StakingTransactionOutput
+StakingSigner::prepareOutput(const Data encoded,
+                             const StakingTransaction<Directive> stakingTx) noexcept {
+    auto protoOutput = Proto::StakingTransactionOutput();
+
+    auto v = store(stakingTx.v);
+    auto r = store(stakingTx.r);
+    auto s = store(stakingTx.s);
+
+    protoOutput.set_encoded(encoded.data(), encoded.size());
+    protoOutput.set_v(v.data(), v.size());
+    protoOutput.set_r(r.data(), r.size());
+    protoOutput.set_s(s.data(), s.size());
+
+    return protoOutput;
+}
+
+template <>
```

Comment:
Is the 'template<>' really needed?

Reply:
It is needed for forcing template specialization